### PR TITLE
Buffs DIY Chainsaw

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Buffs the DIY chainsaw.

This weapon is really weak, all things considered; it *requires* two hands, requires a  decent amount of effort to actually build, and isn't all that much better than a spear (and you can't even throw it).

The damage on it has been increased from 21 to 24.

Other changes:

- Spinning with the double e-sword will make you flip about.

Code changes

- Cleaned up two-handed.dm Also make head spears a bit more sane; there was two ways to make them; attacking a head with a spear or attacking a spear with the head. I removed attacking the head with a spear, as this override the behavior of sharp objects and it's, quite frankly redundant, and inconsistent behavior. Also made it so the spear is stored in the mounted head object instead of created each time.

double-esword spinning

img

:cl: Fox McCloud
tweak: Buffed DIY chainsaws damage slightly
tweak: You now flip about when you spin with a double e-sword
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/twohanded.dm [hunk 1]
```diff
@@ -437,18 +441,18 @@
 	icon_state = "gchainsaw_off"
 	flags = CONDUCT
 	force = 13
-	var/force_on = 21
+	var/force_on = 24
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: Actual change here.

File: code/game/objects/items/weapons/twohanded.dm [hunk 2]
```diff
@@ -226,26 +226,30 @@
...
-			for(var/i in list(1,2,4,8,4,2,1,2,4,8,4,2))
-				user.dir = i
-				sleep(1)
+		INVOKE_ASYNC(src, .proc/jedi_spin, user)
+
+/obj/item/twohanded/dualsaber/proc/jedi_spin(mob/living/user)
+	for(var/i in list(NORTH, SOUTH, EAST, WEST, EAST, SOUTH, NORTH, SOUTH, EAST, WEST, EAST, SOUTH))
+		user.setDir(i)
+		if(i == WEST)
+			user.SpinAnimation(7, 1)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: And here.

File: code/game/objects/items/weapons/twohanded.dm [hunk 3]
```diff
@@ -562,43 +566,41 @@
...
 	if(charged < 5)
 		charged++
-	return
 
 /obj/item/twohanded/singularityhammer/update_icon()  //Currently only here to fuck with the on-mob icons.
 	icon_state = "mjollnir[wielded]"
-	return
-
-/obj/item/twohanded/singularityhammer/proc/vortex(var/turf/pull as turf, mob/wielder as mob)
-	for(var/atom/X in orange(5,pull))
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: Why this looped through every single atom, then did a type check for `/atom/movable` is beyond me. Why not just do built in type checking by making it `/atom/movable` to begin with?

File: code/game/objects/items/weapons/twohanded.dm [hunk 4]
```diff
@@ -562,43 +566,41 @@
...
-		if(istype(X, /atom/movable))
-			if(X == wielder) continue
-			if((X) &&(!X:anchored) && (!istype(X,/mob/living/carbon/human)))
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: then it uses `:` to boot. 😱

File: code/game/objects/items/weapons/twohanded.dm [hunk 5]
```diff
@@ -381,48 +386,47 @@
...
-	return ..()
-
-/obj/item/twohanded/spear/attackby(var/obj/item/I, var/mob/living/user)
+/obj/item/twohanded/spear/attackby(obj/item/I, mob/living/user)
 	if(istype(I, /obj/item/organ/external/head))
-		to_chat(user, "<span class='notice'>You stick the head onto the spear and stand it upright on the ground.</span>")
-		var/obj/structure/headspear/HS = new /obj/structure/headspear(user.loc)
-		var/matrix/M = matrix()
-		I.transform = M
-		usr.drop_item()
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/twohanded.dm]: This code is so bad. Why in the world would you do this. You have a `user` argument, but you use `usr` anyway? What the heck, man.

File: code/game/objects/items/weapons/twohanded.dm [hunk 6]
```diff
@@ -673,27 +673,26 @@
...
-				Z.visible_message("<span class='danger'>[Z.name] was sent flying by a blow from the [src.name]!</span>", \
+				Z.visible_message("<span class='danger'>[Z.name] was sent flying by a blow from the [name]!</span>", \
 					"<span class='userdanger'>You feel a powerful blow connect with your body and send you flying!</span>", \
 					"<span class='danger'>You hear something heavy impact flesh!.</span>")
 				var/atom/throw_target = get_edge_target_turf(Z, get_dir(src, get_step_away(Z, src)))
 				Z.throw_at(throw_target, 200, 4)
 				playsound(user, 'sound/weapons/marauder.ogg', 50, 1)
 			else if(wielded && Z.health < 1)
-				Z.visible_message("<span class='danger'>[Z.name] was blown to peices by the power of [src.name]!</span>", \
+				Z.visible_message("<span class='danger'>[Z.name] was blown to peices by the power of [name]!</span>", \
```
Comment (Citinited) [inline code/game/objects/items/weapons/twohanded.dm]: `pieces` is spelt incorrectly

**Comment by DesolateG** [comment]
can you put up a gif of the flips? that sounds hilarious and all i can picture is Yoda lightsaber battles @Fox-McCloud

**Comment by Fox-McCloud** [comment]
`EDIT: Shit I thought he actually added spinning for the chainsaw instead of upgrading the desword, no bootleg lumberjack gymnastics`

🤔:thinking::thinking::thinking::thinking::thinking:

**Comment by shazbot194** [comment]
> requires a decent amount of effort to actually build

Yes...
image

On a more serious note, I question how effective a 3 damage upgrade will actually be, with the effect seemingly to be at most one less hit, but I do support this PR.

**Comment by Anasari** [comment]
If an item is in the game, it ought to have some reason to be used, in at least some situation. 

A 3 damage difference would differentiate it enough to make it a viable choice.

+1 <3

**Comment by taukausanake** [comment]
> If an item is in the game, it ought to have some reason to be used

I would personally use it for surgery. Kind of wish it had a special interaction, like splattering blood everywhere and making the person scream until the limb was removed. I'd love to see a surgical chainsaw

**Pull request merged**